.github/workflows/build-test.yaml

```diff
@@ -2948,6 +2948,152 @@ jobs:
           aws-secret-access-key: '${{ secrets.E2E_SUPPORT_BUNDLE_AWS_SECRET_ACCESS_KEY }}'
 
 
+  validate-deployment-orchestration:
+    runs-on: ubuntu-20.04
+    needs: [ enable-tests, can-run-ci, build-push-kotsadm-image, build-kurl-proxy, build-migrations, push-minio, push-mc, push-rqlite ]
+    strategy:
+      fail-fast: false
+      matrix:
+        k8s_version: [ v1.27.1-k3s1 ]
+    env:
+      APP_SLUG: deployment-orchestration
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - uses: azure/setup-helm@v3
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - uses: replicatedhq/action-k3s@main
+        with:
+          version: ${{ matrix.k8s_version }}
+          arguments: ${{ startsWith(matrix.k8s_version, 'v1.22') && '--egress-selector-mode disabled' || '' }}
+
+      - name: download kots binary
+        uses: actions/download-artifact@v3
+        with:
+          name: kots
+          path: bin/
+
+      - run: chmod +x bin/kots
+
+      - name: run the test
+        run: |
+          set +e
+          echo ${{ secrets.DEPLOYMENT_ORCHESTRATION_LICENSE }} | base64 -d > license.yaml
+          ./bin/kots \
+            install "$APP_SLUG/automated" \
+            --license-file license.yaml \
+            --no-port-forward \
+            --namespace "$APP_SLUG" \
+            --shared-password password \
+            --kotsadm-registry ttl.sh \
+            --kotsadm-namespace automated-${{ github.run_id }} \
+            --kotsadm-tag 24h
+
+          EXIT_CODE=$?
+          if [ $EXIT_CODE -ne 0 ]; then
+            echo "------pods:"
+            kubectl -n "$APP_SLUG" get pods
+            echo "------kotsadm logs"
+            kubectl logs -l app=kotsadm --tail=100 --namespace "$APP_SLUG"
+            exit $EXIT_CODE
+          fi
+
+          function wait_for_log {
+            local counter=1
+            local timeout=30
+            local log_pattern="$1"
+
+            while ! kubectl logs deploy/kotsadm -c kotsadm -n "$APP_SLUG" | grep -i "$log_pattern"; do
+              ((counter += 1))
+              if [ $counter -gt $timeout ]; then
+                echo "Timed out waiting for log $log_pattern"
+                exit 1
+              fi
+              sleep 1
+            done
+          }
+
+          wait_for_log "applying phase -9999"
+
+          wait_for_log "applying phase -3"
```

Comment:
i'm being picky, and i don't have a solution off the top of my head, but if the API applies -3 _then_ -9999, this test wouldn't catch that, because both logs would be present when checking for the -9999 log. same for https://github.com/replicatedhq/kots/pull/3903/files#diff-bd4e577d8cce18da4cecc900fca127f9a0d57d706c9cdb3507b778c1d574a54fR3075-R3079

Reply:
now that i finished reviewing the PR, I see that you already added a unit test for validating the ordering of phases, so it's not necessary to cover it here.